src/go/wsl-helper/pkg/dnsupdater/run.go

```diff
@@ -0,0 +1,92 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
+
+*/
+package dnsupdater
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"net/netip"
+	"os"
+	"sort"
+
+	"golang.org/x/sys/windows"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+func Run(path string) error {
+	addrs, err := winipcfg.GetAdaptersAddresses(windows.AF_UNSPEC, winipcfg.GAAFlagIncludeAll)
```

Comment:
Any particular reason we're not using `AF_INET` here? We don't care about IPv6, as far as I can tell? It makes some of the other comments moot.

Reply:
Well, it's more that later on we look for interfaces with IPv4 metric _anyway_, so it's not like getting IPv6 interfaces would have helped…